application/client.js

```diff
@@ -152,7 +152,14 @@ module.exports = HttpGateway.extend({
                     try {
                         responseData = JSON.parse(responseText);
                     } catch (e) {
-                        responseData = responseText;
```

Comment:
It looks like (from what digging I did) the XmlHttpRequest automagically does an OPTIONS request for CORS purposes.  The request itself is done on a lower level library but does not check for an invalid response or XHR result.

It might actually be better to build the raw XHR ourselves, especially since we're now allowing files.  The way it's currently implemented limits us to one file upload with no extra parameters in the body.